demo/lib/distro.bash

```diff
@@ -487,11 +488,29 @@ EOF
 if grep -q NAME=Fedora /etc/os-release; then
     if ! grep -q systemd.unified_cgroup_hierarchy=0 /proc/cmdline; then
         sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
-        shutdown -r now
+        reboot_needed=1
     fi
 fi
 EOF
     fi
+
+    # Using swapoff is not enough as we also need to disable the swap from systemd
+    # and then reboot the VM
+    cat <<EOF
+if swapon --show | grep -q partition; then
+    sed -E -i '/^\\/.*[[:space:]]swap[[:space:]].*\$/d' /etc/fstab
+    systemctl --type swap
```

Comment:
nitpick: what is the purpose of this `systemctl --type swap` here ?

Reply:
I added it there in order to see what the system thinks about the swap devices, so just a debugging aid if something goes wrong. Can be removed if you think that is not needed.

Reply:
No need to remove it if you consider it useful. I just wanted to make sure it is not an accidental leftover.